src/components/QueryEditor/VisualQueryEditor/FilterItem.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import { useAsyncFn } from 'react-use';
+
+import { SelectableValue, toOption } from '@grafana/data';
+import { AccessoryButton, AutoSizeInput, InputGroup, Select } from '@grafana/ui';
+
+import { AdxDataSource } from '../../../datasource';
+import { AdxColumnSchema, KustoQuery } from '../../../types';
+import {
+  QueryEditorExpressionType,
+  QueryEditorOperatorExpression,
+} from 'components/LegacyQueryEditor/editor/expressions';
+import {
+  getOperatorExpressionOptions,
+  getOperatorExpressionValue,
+  setOperatorExpressionName,
+  setOperatorExpressionProperty,
+  setOperatorExpressionValue,
+} from './utils/utils';
+import { isMulti, toOperatorOptions } from './utils/operators';
+import { columnsToDefinition, valueToDefinition } from 'schema/mapper';
+import { QueryEditorPropertyType } from 'schema/types';
+
+interface FilterItemProps {
+  datasource: AdxDataSource;
+  query: KustoQuery;
+  filter: Partial<QueryEditorOperatorExpression>;
+  columns: AdxColumnSchema[] | undefined;
+  templateVariableOptions: SelectableValue<string>;
+  onChange: (item: QueryEditorOperatorExpression) => void;
+  onDelete: () => void;
+}
+
+const FilterItem: React.FC<FilterItemProps> = (props) => {
+  const { datasource, query, filter, onChange, onDelete, columns, templateVariableOptions } = props;
+
+  const loadValues = async () => {
+    if (!filter.property?.name) {
+      return [];
+    }
+
+    return datasource
+      .autoCompleteQuery(
+        {
+          ...query,
+          search: {
+            operator: { name: 'isnotempty', value: '' },
+            property: { name: filter.property?.name, type: filter.property.type },
+            type: QueryEditorExpressionType.Operator,
+          },
+        },
+        columns
+      )
+      .then((result: string[]) => {
+        if (Array.isArray(templateVariableOptions.options) && filter.property?.type === 'string') {
+          // When selecting a string, automatically quote template variables to generate a valid syntax
+          templateVariableOptions.options = templateVariableOptions.options.map((op: SelectableValue<string>) => {
+            return { label: op.value, value: `'${op.value}'` };
+          });
+        }
+        return result.map((r): SelectableValue<string> => ({ label: r, value: r })).concat(templateVariableOptions);
```

Comment:
this logic return suggestions for a filter. In the `search` expression yhou can see that we are trying to find properties that are not empty for a certain value (`property.name`). It append template variables to the final result.